x-pack/legacy/plugins/lens/public/indexpattern_datasource/datapanel.tsx

```diff
@@ -481,39 +480,43 @@ export const InnerIndexPatternDataPanel = function InnerIndexPatternDataPanel({
-                            defaultMessage:
-                              'No fields match the current filters. Try changing your filters or time range.',
-                          })
-                        : i18n.translate('xpack.lens.indexPatterns.noFieldsLabel', {
-                            defaultMessage: 'No fields exist in this index pattern.',
-                          })
+                    localState.typeFilter.length || localState.nameFilter.length
+                      ? i18n.translate('xpack.lens.indexPatterns.noFilteredFieldsLabel', {
+                          defaultMessage: 'No fields match the current filters.',
+                        })
+                      : showEmptyFields
+                      ? i18n.translate('xpack.lens.indexPatterns.noFieldsLabel', {
+                          defaultMessage: 'No fields exist in this index pattern.',
+                        })
                       : i18n.translate('xpack.lens.indexPatterns.emptyFieldsWithDataLabel', {
-                          defaultMessage:
-                            'No fields have data with the current filters and time range. Try changing your filters or time range.',
+                          defaultMessage: 'Looks like you don’t have any data.',
                         })
                   }
                 >
                   {(!showEmptyFields ||
                     localState.typeFilter.length ||
                     localState.nameFilter.length) && (
-                    <EuiButtonEmpty
-                      size="xs"
-                      color="primary"
-                      flush="left"
-                      data-test-subj="lnsDataPanelShowAllFields"
-                      onClick={() => {
-                        trackUiEvent('indexpattern_show_all_fields_clicked');
-                        clearLocalState();
-                        onToggleEmptyFields(true);
-                      }}
-                    >
-                      {i18n.translate('xpack.lens.indexPatterns.showAllFields.buttonText', {
-                        defaultMessage: 'Show all fields',
-                      })}
-                    </EuiButtonEmpty>
+                    <>
+                      <strong>
+                        {i18n.translate('xpack.lens.indexPatterns.noFields.tryText', {
+                          defaultMessage: 'Try:',
+                        })}
+                      </strong>
+                      <ul>
+                        <li>
+                          {i18n.translate('xpack.lens.indexPatterns.noFields.extendTimeBullet', {
+                            defaultMessage: 'Extending the time range',
+                          })}
+                        </li>
+                        <li>
+                          {i18n.translate('xpack.lens.indexPatterns.noFields.fieldFilterBullet', {
+                            defaultMessage:
+                              'Using the field filter {arrow} to show fields without data',
```

Comment:
It wasn't immediately clear to me "field filter" was referring to "Filter by type" - I think this could get easily confused with the "Add filter" button from the filter bar (especially since it's right above it). What about `Using the type filter...` (same word as in the label)?

Reply:
I like the `Filter fields by type` option. I'll go with that

Reply:
I like:

Using the Filter by type option to show fields without data

It matches the UI and it doesn't include the word "field" twice.

The rest of the text LGTM.

Reply:
Done, updated summary screenshots. Thanks @gchaps